### PR TITLE
mi: fix the rc for nvme_mi_scan_ep

### PR DESCRIPTION
**Pull request by jinliangw**
nvme_mi_scan_ep should return error code of calling function.
This will help the upper layer know the reason of failure.

### CODE REVIEW

File: src/nvme/mi.c [hunk 1]
```diff
@@ -327,7 +327,7 @@ int nvme_mi_scan_ep(nvme_mi_ep_t ep, bool force_rescan)
 
 	rc = nvme_mi_mi_read_mi_data_ctrl_list(ep, 0, &list);
 	if (rc)
-		return -1;
+		return rc;
```
Comment (igaw) [inline src/nvme/mi.c:330]: The library uses the errno style error handling approach. So the return code `-1` is correct. But I see that `nvme_mi_mi_read_mi_data_ctrl_list` is not following this approach. Thus I think you need to update that code.

**Comment by igaw** [comment]
@jk-ozlabs

File: src/nvme/mi.c (hunk 1 above)
Comment (igaw) [reply src/nvme/mi.c:330]: Indeed :)

File: src/nvme/mi.c [hunk 2]
```diff
@@ -700,7 +696,17 @@ int nvme_mi_admin_admin_passthru(nvme_mi_ctrl_t ctrl, __u8 opcode, __u8 flags,
 		resp.data_len = data_len;
 	}
 
+	if (timeout_ms != 0) {
+		/* When user forgets to set timeout_ms, the timeout_ms will be 0.
```
Comment (jk-ozlabs) [inline src/nvme/mi.c:700]: The logic here is sound, but you're documenting the timeout == 0 case in the condition where the timeout is not zero.

Perhaps something like:

```
    /* if the user has specified a custom timeout, save the current timeout and override */
```

\- it's not necessarily an error ("user forgets") to pass zero here.

File: src/nvme/mi.c [hunk 3]
```diff
@@ -638,6 +638,7 @@ int nvme_mi_admin_admin_passthru(nvme_mi_ctrl_t ctrl, __u8 opcode, __u8 flags,
 	struct nvme_mi_admin_req_hdr req_hdr;
 	struct nvme_mi_resp resp;
 	struct nvme_mi_req req;
+	unsigned int ori_timeout;
```
Comment (jk-ozlabs) [inline src/nvme/mi.c:641]: Super minor, but maybe call this `timeout_save`

**Review by jk-ozlabs — COMMENTED** [review]
Code looks good, I'd just suggest a couple of minor doc/naming changes.

As a more general (later) change, should we pass the timeout into `nvme_mi_submit`, rather than doing the save & restore?

File: src/nvme/mi.c (hunk 2 above)
Comment (jinliangw) [reply src/nvme/mi.c]: Done.

**Review by jk-ozlabs — APPROVED** [review]
LGTM, thanks!

**Comment by igaw** [comment]
Thanks!

**Pull request merged**